Correction of `gradle_command` string value

ali-sajjad-rizavi opened the pull request:
Since you are already running the `cd_command` to go inside the project directory, there should not be `project_path` before "gradlew".
I was having `System cannot find the file specified` issue when I was using it as it was. So, I changed the library code to generate the APK.
BEFORE:
```
cd /path/to/project
/path/to/project/gradlew assembleRelease
```

AFTER:
```
cd /path/to/project
gradlew assembleRelease
```
Corrected!

ali-sajjad-rizavi committed:
Correction of `gradle_command` string value

Since you are already running the `cd_command` to go inside the project directory, there should not be `project_path` before "gradlew".
I was having `System cannot find the file specified` issue when I was using it as it was. So, I changed the library code to generate the APK.
BEFORE:
```
cd /path/to/project
/path/to/project/gradlew assembleRelease
```

AFTER:
```
cd /path/to/project
gradlew assembleRelease
```
Corrected!

osfunapps commented:
Hey,

Can you please write which function you called and with which arguments?

ali-sajjad-rizavi commented:
I had called the `build_apk` function according to the instructions provided in `README.md`. I also discovered something else to make it possible to also run on Unix terminal. For Unix based terminals, we need to use `./gradlew` instead of `gradle`. Will let you know soon! This project is very useful!

osfunapps commented:
Thanks for your input.

As written in the docstring of `build_apk()` and `build_app_bundle()`:
> gradle_path: the path to your gradle file (if left blank, will use the gradle wrapper in the project).

It means that be default you don't need to set anything in the `build_gradle` argument.

I've added a clearer explanation on the subject cause it probably wasn't clear enough. 
You can read about it in the [readme here](https://github.com/osfunapps/os_android_apk_builder-py#alternative-gradle-path).

Also, note that the Gradle Wrapper in your project is different from the Gradle you install on your system (it could be more updated or less, depends on your project).

If something isn't clear please let me know,
Oz.

ali-sajjad-rizavi commented:
So I notice that you have updated the `_general_utils.py` 1 hour ago, and corrected your `gradle_path` variable yourself!
But another issue comes up now:
```
WARNING:[APK Builder]: *** NOTICE: build.gradle file has the line: "signingConfig signingConfigs.release". If you want to use KeyStoreProperties, remove this line from your build.gradle
INFO:[APK Builder]: Preparing build.gradle file for automated release...
INFO:[APK Builder]: Clearing latest .apk file from project...
INFO:[APK Builder]: Building the apk...
'.' is not recognized as an internal or external command,
operable program or batch file.
Traceback (most recent call last):
  File "C:\Users\Ali\Desktop\testapkbuilder\bot.py", line 20, in <module>
    apk_builder.build_apk(project_path='C:/Users/Ali/Desktop/BCSF17M555_TermProject/SuperAnimeDownloader_android',
  File "C:\Users\Ali\AppData\Local\Programs\Python\Python39\lib\site-packages\os_android_apk_builder\apk_builder.py", line 40, in build_apk
    return _build(project_path, apk_dst_dir_path, key_store_properties, version_properties, gradle_path, _res.APK, remove_older_apks_from_output_dir)
  File "C:\Users\Ali\AppData\Local\Programs\Python\Python39\lib\site-packages\os_android_apk_builder\apk_builder.py", line 102, in _build
    raise Exception(f"ERROR: Failed to create {binary_type} file")
Exception: ERROR: Failed to create apk file
```
Here the gradle command becomes `./gradlew assembleRelease`, and `./` causes error on cmd:
```
'.' is not recognized as an internal or external command,
operable program or batch file.
```

### The code I used according to your `README`:
```
from os_android_apk_builder.objs.KeyStoreProperties import KeyStoreProperties
from os_android_app_version_changer.objs.VersionProperties import VersionProperties


# set your KeyStore properties
key_store_props = KeyStoreProperties(key_store_file_path='C:/Users/Ali/Desktop/sample-keystore.jks',
                                     store_password='abcd1234',
                                     key_alias='sample-alias',
                                     key_password='abcd1234',
                                     v1_signing_enabled=True,
                                     v2_signing_enabled=True)

# set the version properties (version code and version name)
version_props = VersionProperties(new_version_code=VersionProperties.RAISE_VERSION_BY_ONE,
new_version_name="1.0.3")


from os_android_apk_builder import apk_builder

apk_builder.build_apk(project_path='C:/Users/Ali/Desktop/BCSF17M555_TermProject/SuperAnimeDownloader_android',
                      apk_dst_dir_path='C:/Users/Ali/Desktop/BCSF17M555_TermProject',
                      key_store_properties=key_store_props,
                      version_properties=version_props)
```

ali-sajjad-rizavi committed:
Fixed `gradle_path` string for windows

After you updated the `_general_utils.py`, this is the new change I would like to suggest.
If the `gradle_path` is `./gradlew` (current directory), then this will work on Unix based system terminal only. Windows cmd shows this error:
```
'.' is not recognized as an internal or external command,
operable program or batch file.
```
To fix this, I added this line: `gradle_path = '.\\gradlew' if gradle_path == './gradlew' and os.name == 'nt' else gradle_path`
For more clarity:
```
# if OS is windows the gradle_path is current directory
if os.name == 'nt' and gradle_path == './gradlew':
    gradle_path = '.\\gradlew' # Using '\' character instead of '/'
```

ali-sajjad-rizavi committed:
Update _general_utils.py

ali-sajjad-rizavi commented:
I updated the file with another suggestion. You can check it out!
Only this line added for the fix: `gradle_path = '.\\gradlew' if gradle_path == './gradlew' and os.name == 'nt' else gradle_path`

osfunapps commented:
Great catch! Thank you

osfunapps commented:
Notice that I've added a path.join() to make the request cross platform efficient and changed the build_gradle path to None to reflect the changes in each platform.

In short, it was pretty confusing for users to set their gradle wrapper so now they just need to pass None (or pass nothing at all) if they want to use the wrapper.

ali-sajjad-rizavi commented:
Yes! That's much better readable choice!
